ctapipe/core/provenance.py

```diff
@@ -156,9 +174,8 @@ def activity(self, name):
     @property
     def current_activity(self):
```

Comment:
It is not clear to me why the code got moved from here to the entirely new `_get_current_or_start_activity`?

Within `provenance.py` itself it looks like the new implementation is just adding a new level if indirection but will otherwise execute in exactly the same way, and the actual fix of the bug in `datawriter.py` does not use this new function either.

Reply:
The difference is now we only do that in the methods that attach new information (add_config, add_input, add_output), not in the ones that just access the current activity for reading.

The problem was that the `DataWriter` just by accessing the `current_activity` would create a fresh one and write the info of that and not write the info of the last finished activity.